mbed_devices/_internal/windows/windows_identifier.py

```diff
@@ -0,0 +1,68 @@
+"""Defines a Windows identifier."""
+from typing import NamedTuple, Any, Optional
+
+from mbed_devices._internal.windows.component_descriptor_utils import is_undefined_value
+
+INSTANCE_ID_CHARACTER = "&"
+
+
+class WindowsUID(NamedTuple):
```

Comment:
I think we should drop the `NamedTuple` inheritance if we're implementing `__eq__` and `__hash__` anyway, here is a nice breakdown of why we should probably avoid inheriting `NamedTuple` https://www.attrs.org/en/stable/why.html#namedtuples

Reply:
maybe we should consider the use of `attrs` in the future but I would rather discuss about this as part of a more general decision

Reply:
I'm not sure if we should use `attrs`. However, I do think for the reasons mentioned in the link, and due to the fact we define our own `__eq__`, `__repr__` etc, we shouldn't inherit from `NamedTuple`.